torch_tvm/compiler.cpp

```diff
@@ -352,37 +358,37 @@ void TVMCompiler::run(Stack& stack) {
   // Parameters are managed by cached tvm_param_tensors.
   // They get deallocated when cache_ is deleted.
   std::vector<DLManagedTensorPtr> dl_tensor_list;
-  for (auto i = 0; i < cache_[spec].input_values.size(); ++i) {
-    auto* value = cache_[spec].input_values[i].first;
-    bool input_immutable = cache_[spec].input_values[i].second;
+  for (auto& input_value : cache_[spec].input_values) {
+    auto* value = input_value.first;
+    TVMGraphInputInfo& graph_input = input_value.second;
     if (!value_to_ivalue.count(value)) {
       auto optional_ivalue = toIValue(value);
       TORCH_INTERNAL_ASSERT(optional_ivalue.has_value());
       value_to_ivalue[value] = optional_ivalue.value();
     }
-    auto ivalue = value_to_ivalue.at(cache_[spec].input_values[i].first);
+    auto ivalue = value_to_ivalue.at(value);
     //auto tensor = ivalue.toTensor().to(at::kFloat);
     auto tensor = ivalue.toTensor();
     DLManagedTensor* dl_tensor;
     if (tensor.is_contiguous() &&
```

Comment:
Can we make a function out of this chunk?

Reply:
Let me see if this is possible to do. Position of the node for a given variable name is known by the runtime. I am not sure if this is exposed, but I will look into it.

Reply:
Sure.